music21/musicxml/m21ToXml.py

```diff
@@ -3265,55 +3280,88 @@ def parseFlatElements(
 
         self.currentVoiceId = voiceId
 
-        # group all objects by offsets and then do a different order than normal sort.
-        # that way chord symbols and other 0-width objects appear before notes as much as
+        # We order things differently if there are any SpannerAnchors in the flat stream.
+        # If there are SpannerAnchors, we do two passes. The first pass skips all
+        # prePostObjectSpanners processing, and then the second pass goes back to the
+        # beginning and does all the prePostObjectSpanners processing.
+        # If there are no SpannerAnchors, we do one pass, doing any prePostObjectSpanners
+        # processing as we process each element.
+
+        hasSpannerAnchors: bool = bool(m[spanner.SpannerAnchor])
```

Comment:
since this is already a flat stream, `bool(m.getElementsByClass(spanner.SpannerAnchor))` is more efficient.

Reply:
done